### PR TITLE
Rework network configuration

### PR DESCRIPTION
**Pull request by rtsisyk**
- Replace `bind_ip` and `bind_port` with `endpoint`.
- Add `STEGOS_NETWORK_ENDPOINT` environment variable.
- Replace `advertised_addresses` with `public_endpoint`.
- Add `STEGOS_PUBLIC_NETWORK_ENDPOINT` environment variable.
- Use `ipaddr:port` format for `seed_nodes`.
- Improve logging.

### CODE REVIEW

File: src/bin/stegosd/main.rs [hunk 1]
```diff
@@ -369,6 +389,32 @@ fn run() -> Result<(), Error> {
...
+                    SocketAddr::from_str(&uri)
+                        .map(|_| ())
+                        .map_err(|e| format!("{}", e))
+                })
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("public-network-endpoint")
+                .long("public-network-endpoint")
+                .env("STEGOS_PUBLIC_NETWORK_ENDPOINT")
```
Comment (rtsisyk) [inline src/bin/stegosd/main.rs]: I don't like this long name. Any better ideas? What is about `STEGOS_NETWORK_LISTEN`, `STEGOS_NETWORK_PUBLISH`?

File: network/src/libp2p_network/mod.rs [hunk 2]
```diff
@@ -180,10 +183,11 @@ fn new_service(
         Swarm::new(transport, behaviour, peer_id)
     };
 
-    let mut bind_ip = Multiaddr::from(Protocol::Ip4(config.bind_ip.clone().parse()?));
-    bind_ip.push(Protocol::Tcp(config.bind_port));
-
-    Swarm::listen_on(&mut swarm, bind_ip).unwrap();
+    if config.endpoint != "" {
```
Comment (rtsisyk) [inline network/src/libp2p_network/mod.rs]: Please note that I don't start listener if network.endpoint is empty. I tested on my local machine and it works.

File: src/bin/stegosd/main.rs (hunk 1 above)
Comment (vldm) [reply src/bin/stegosd/main.rs]: why you need public world in variable name?
LISTEN_ADDR should be ok as well.

File: network/src/libp2p_network/mod.rs (hunk 2 above)
Comment (vldm) [reply network/src/libp2p_network/mod.rs]: Maybe we should add log entry, with info that user has lack of some features.

File: network/src/ncp/behavior.rs [hunk 3]
```diff
@@ -88,25 +90,32 @@ pub struct Ncp<TSubstream> {
...
-            .filter_map(|a| match a.parse() {
-                Ok(addr) => Some(addr),
-                Err(_) => None,
-            })
+            .map(|a| socket_to_multi_addr(&SocketAddr::from_str(a).expect("Invalid seed_nodes")))
             .collect();
+
+        let public_endpoint = if config.public_endpoint != "" {
+            let endpoint =
+                SocketAddr::from_str(&config.public_endpoint).expect("Invalid public_endpoint");
```
Comment (vldm) [inline network/src/ncp/behavior.rs]: Would `SocketAddr::from_str` support hostnames?

Comment (rtsisyk) [reply network/src/ncp/behavior.rs]: I don't think so. DNS support is a different story.

File: src/bin/stegosd/main.rs (hunk 1 above)
Comment (rtsisyk) [reply src/bin/stegosd/main.rs]: Discussed.

**Pull request merged**